packages/selection/frontend/sirius-components-selection/src/SelectionDialog.types.ts

```diff
@@ -18,3 +18,14 @@ export interface SelectionDialogProps {
   onClose: () => void;
   onFinish: (selectedObjectId: string) => void;
 }
+
+export interface DiagramDialogState {
+  treeDescriptionId: string;
+  message: string;
```

Comment:
string or null to properly identify that you don't have the relevant data. Your code only works because ```if(aVariableContainingAnEmptyString)``` evaluates to `false` and it does not enter the `if`. That's a very narrow assumption that empty string equals false.

Reply:
And you probably don't need this in the state too

Reply:
The state and the useEffect were indeed not needed